### PR TITLE
Protocol bug

### PR DESCRIPTION
**Pull request by kosssi**
**functional:**
- [x] fix 404 error on the first replication
- [x] fix authentication error when it download a file
- [x] Add 4 minutes for background service execution
- [x] Fix error without message
- [x] Fix synchronization date in configuration page when I click on synchronization button
- [x] Add a state to restart application after login page when it's logged
- [x] Add a state to restart application after configuration page when it's configured
- [x] Add a state to restart application after first replication page when is terminated
- [x] Synchronize data even if battery informations is undefined

**technique:**
- [x] move www-src to src
- [x] create new configuration file
- [x] update logs
- [x] create class for database
- [x] create class for all cozy request
- [x] upgrade version
- [ ] Add tests for new code
- [ ] Test new migration

### CODE REVIEW

File: src/app/views/layout.coffee [hunk 1]
```diff
@@ -205,7 +205,8 @@ module.exports = class Layout extends BaseView
 
     showError: (error) =>
         log.debug 'showError'
-        @errorIndicator.text t error.message
+        msg = error.message or error.reason
```
Comment (kosssi) [inline src/app/views/layout.coffee]: Fix error without message

File: src/app/replicator/main.coffee [hunk 2]
```diff
@@ -492,6 +482,9 @@ module.exports = class Replicator extends Backbone.Model
         @set 'inSync', true
         @_sync options, (err) =>
             @set 'inSync', false
+            # Skip first synchronisation
+            # todo: find better solution
+            callback() if err?.status is 404
```
Comment (kosssi) [inline src/app/replicator/main.coffee]: fix 404 error on the first replication

File: src/app/replicator/filesystem.coffee [hunk 3]
```diff
@@ -141,7 +141,6 @@ module.exports.download = (options, progressback, callback) ->
     ]
 
     {url, path, auth} = options
-    url = encodeURI url
```
Comment (kosssi) [inline src/app/replicator/filesystem.coffee]: fix authentication error when it download a file

File: src/app/background_service.coffee [hunk 4]
```diff
@@ -80,9 +58,9 @@ module.exports = BackgroundService =
...
 
             finally
-                # "Watchdog" : in all cases, kill service after 10'
+                # "Watchdog" : in all cases, kill service after 14'
                 setTimeout ->
                     # call this javabinding directly on object to avoid
                     # Error 'NPMethod called on non-NPObject'
                     window.service.workDone()
-                , 10 * 60 * 1000
+                , 14 * 60 * 1000
```
Comment (kosssi) [inline src/app/background_service.coffee]: Add 4 minutes for background service execution

File: src/app/views/menu.coffee [hunk 5]
```diff
@@ -26,6 +25,7 @@ module.exports = class Menu extends BaseView
     backup: ->
         app.layout.closeMenu()
         app.init.launchBackup()
+        app.router.forceRefresh()
```
Comment (kosssi) [inline src/app/views/menu.coffee]: Fix synchronization date in configuration page when I click on synchronization button

File: src/app/init.coffee [hunk 6]
```diff
@@ -321,49 +325,56 @@ module.exports = class Init
...
-            # First start error
-            'goTofDeviceName': 'f1QuitSplashScreen'
-            'goTofInitContacts': 'f2QuitSplashScreen'
-            'goTofInitCalendars': 'f3QuitSplashScreen'
-            'goTofUpdateIndex': 'f4QuitSplashScreen'
+        'aInitFileSystem': 'fileSystemReady': 'aQuitSplashScreen'
+        'aQuitSplashScreen': 'viewInitialized': 'aApplicationState'
+        'aApplicationState':
+            'default': 'fWizardWelcome'
+            'deviceCreated': 'fWizardFiles'
```
Comment (kosssi) [inline src/app/init.coffee]: Add a state to restart application after login page when it's logged

File: src/app/init.coffee [hunk 7]
```diff
@@ -321,49 +325,56 @@ module.exports = class Init
...
+            'appConfigured': 'fFirstSyncView'
```
Comment (kosssi) [inline src/app/init.coffee]: Add a state to restart application after configuration page when it's configured

File: src/app/init.coffee [hunk 8]
```diff
@@ -321,49 +325,56 @@ module.exports = class Init
...
+            'syncCompleted': 'nLoadFilePage'
```
Comment (kosssi) [inline src/app/init.coffee]: Add a state to restart application after first replication page when is terminated

File: src/app/lib/device_status.coffee [hunk 9]
```diff
@@ -70,7 +70,8 @@ module.exports.checkReadyForSync = checkReadyForSync = (callback)->
         setTimeout () ->
             if timeout
                 # We reached timeout, and a batterystatus event hasn't fired yet
-                callbackWaiting new Error "No battery informations"
+                log.warn new Error "No battery informations"
+                callbackWaiting null, true
```
Comment (kosssi) [inline src/app/lib/device_status.coffee]: Synchronize data even if battery informations is undefined

File: src/app/application.coffee [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+            msg += "\n #{t('error try restart')}"
+            alert msg
+        navigator.app.exitApp()
+
+    addDeviceListener: ->
+        log.debug "addDeviceListener"
+
+        document.addEventListener 'deviceready', =>
+            @initialize()
+        , false
```
Comment (nono) [inline src/app/application.coffee]:
```coffee
document.addEventListener 'deviceready', @initialize, false
```

should be enough

**Comment by nono** [comment]
Looks good to me. Do you want to fix the last 2 items of your checklist before merging it?

**Comment by kosssi** [comment]
@nono No you can merge it after test is ok.

**Comment by kosssi** [comment]
thanks

**Pull request merged**